src/libxtp/threecenter_gwbse.cc

```diff
@@ -149,16 +200,127 @@ void TCMatrix_gwbse::FillBlock(std::vector<Eigen::MatrixXd>& block,
       }
     }  // gamma-loop
   }    // alpha-loop
-  for (int k = 0; k < auxshell.getNumFunc(); ++k) {
+
+  return symmstorage;
+}
+
+/*
+ *  Convolution of the GW shell with ALL functions with the DFT orbital
+ * coefficients
+ */
+std::vector<Eigen::MatrixXd> TCMatrix_gwbse::FillBlock(
+    const std::vector<Eigen::MatrixXd>& symmstorage,
+    const Eigen::MatrixXd& dft_orbitals, const AOShell& shell) const {
```

Comment:
you do not need the `shell` as an argument hereor in the CUDA block, because `shell.getNumFunc()` is `symmstorage,size`

Reply:
thanks!